cpp/src/join/hash_join.cu

```diff
@@ -349,11 +349,15 @@ std::size_t hash_join::hash_join_impl::inner_join_size(cudf::table_view const& p
   CUDF_FUNC_RANGE();
   CUDF_EXPECTS(_hash_table, "Hash table of hash join is null.");
 
-  auto build_table = cudf::table_device_view::create(_build, stream);
-  auto probe_table = cudf::table_device_view::create(probe, stream);
+  auto flattened_probe = structs::detail::flatten_nested_columns(
+    probe, {}, {}, structs::detail::column_nullability::FORCE);
+  auto const flattened_probe_table = std::get<0>(flattened_probe);
+
+  auto build_table             = cudf::table_device_view::create(_build, stream);
+  auto d_flattened_probe_table = cudf::table_device_view::create(flattened_probe_table, stream);
```

Comment:
Since we add the `d_` prefix to the latter, we should also add it to the former to enforce consistency:
```suggestion
  auto d_build_table             = cudf::table_device_view::create(_build, stream);
  auto d_flattened_probe_table = cudf::table_device_view::create(flattened_probe_table, stream);
```


Another way (which makes more sense) is to add a suffix `_ptr` instead, because `table_device_view::create` returns a pointer:
```suggestion
  auto build_table_ptr             = cudf::table_device_view::create(_build, stream);
  auto flattened_probe_table_ptr = cudf::table_device_view::create(flattened_probe_table, stream);
```

Reply:
I went with the latter suggestion here.